code/control/DeployPlanDispatcher.php

```diff
@@ -290,6 +285,16 @@ protected function getGitPrevDeploys($project) {
 	 * @return SS_HTTPResponse
 	 */
 	protected function getAPIResponse($output, $statusCode) {
+		// GET and HEAD requests should not change state and therefore
```

Comment:
@mateusz after our discussion about the racy condition of GET and POST at the same time, I've added a check below to ensure only non GET and HEAD requests resets the token.